### ISSUE TITLE
What happens when an import matches two or more ImportOrder groups?

### ISSUE THREAD

**Issue by JayBazuzi** [checkstyle/checkstyle#15626]
Reference documentation for `groups` here: https://checkstyle.sourceforge.io/checks/imports/importorder.html#Properties, which currently explains what happens when an import does not match any group, but it doesn't explain what happens if it matches two or more groups.

For example, if I configure checkstyle with:

```xml
<module name="ImportOrder">
    <property name="groups" value="
        java.,
        com.,
        com.mycompany.,
    "/>
</module>
```

checkstyle will approve this source code:

```java
import java.util.Date;

import com.sun.jna.Platform;

import com.mycompany.Foo;
```

Note that even though both the 2nd and 3rd imports match `com.`, the 3rd import goes in its own group. I can see in the code that checkstyle tries to pick the "best" match, but I wish that was clear in the documentation.

**Comment by rnveach** [comment]
> checkstyle will approve this source code:

It actually does not.

````
$ cat TestClass.java
import java.util.Date;

import com.sun.jna.Platform;

import com.mycompany.Foo;

$ cat TestConfig.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="UTF-8"/>
    <property name="severity" value="warning"/>
    <property name="haltOnException" value="false"/>

    <module name="TreeWalker">
<module name="ImportOrder">
    <property name="groups" value="
        java.,
        com.
        com.mycompany.
    "/>
</module>
    </module>
</module>

$ java -jar checkstyle-10.18.1-all.jar -c TestConfig.xml TestClass.java
Starting audit...
[WARN] TestClass.java:3:1: Extra separation in import group before 'com.sun.jna.Platform' [ImportOrder]
[WARN] TestClass.java:5:1: Extra separation in import group before 'com.mycompany.Foo' [ImportOrder]
[WARN] TestClass.java:5:1: Wrong order for 'com.mycompany.Foo' import. [ImportOrder]
Audit done.
````

**Comment by JayBazuzi** [comment]
You're right. Updated the original post to include the missing commas.
